pandas/tests/test_categorical.py

```diff
@@ -4549,6 +4549,13 @@ def test_concat_categorical(self):
                             'h': [None] * 6 + cat_values})
         tm.assert_frame_equal(res, exp)
 
+    def test_rank_categorical(self):
```

Comment:
Can you add a test with `NaN`s? IIRC these are stored as `-1` for the codes, but should be `NaN` in the output of rank.